crates/sui-core/src/transaction_driver/effects_certifier.rs

```diff
@@ -109,117 +121,73 @@ impl EffectsCertifier {
-                        details,
-                    } => {
-                        // All error cases are retryable until max attempt due to the chance
-                        // of the status being returned from a byzantine validator.
-                        if let Some(details) = details {
-                            return Ok((effects_digest, details));
-                        } else {
-                            if attempts >= MAX_ATTEMPTS {
-                                return Err(TransactionDriverError::ExecutionDataNotFound(
-                                    tx_digest.to_string(),
-                                ));
-                            }
-                            tracing::debug!("Execution data not found, retrying...");
-                        }
-                    }
-                    WaitForEffectsResponse::Rejected { ref reason } => {
-                        if attempts >= MAX_ATTEMPTS {
-                            return Err(TransactionDriverError::TransactionRejected(
-                                reason.to_string(),
-                            ));
-                        }
-                        tracing::debug!("Transaction rejected, retrying... Reason: {}", reason);
-                    }
-                    WaitForEffectsResponse::Expired { epoch, round } => {
-                        if attempts >= MAX_ATTEMPTS {
-                            return Err(TransactionDriverError::TransactionStatusExpired);
-                        }
-                        tracing::debug!(
-                            "Transaction status expired at epoch {}, round {}, retrying...",
-                            epoch,
-                            round.unwrap_or(0),
-                        );
-                    }
-                },
-                Ok(Err(e)) => {
-                    if attempts >= MAX_ATTEMPTS {
-                        return Err(TransactionDriverError::RpcFailure(
-                            name.concise().to_string(),
-                            e.to_string(),
-                        ));
-                    }
-                    tracing::debug!(
-                        "Full effects request failed from {}: {}, retrying...",
-                        name.concise(),
-                        e
-                    );
-                }
-                Err(_) => {
-                    if attempts >= MAX_ATTEMPTS {
-                        return Err(TransactionDriverError::TimeoutGettingFullEffects);
+        match timeout(
+            WAIT_FOR_EFFECTS_TIMEOUT,
+            client.wait_for_effects(raw_request.clone(), options.forwarded_client_addr),
+        )
+        .await
+        {
+            Ok(Ok(response)) => match response {
+                WaitForEffectsResponse::Executed {
+                    effects_digest,
+                    details,
+                } => {
+                    // All error cases are retriable until max attempt due to the chance
```

Comment:
We won't retry waiting for effects on errors on a per validator basis for the same consensus position that are non-retriable. i.e. expired, rejected, etc.

Reply:
Removed the comment.